### ISSUE TITLE
V2.15.0: Failed to parse PDF - Could not reproduce the invoice..

### ISSUE THREAD

**Issue by schuettef** [ZUGFeRD/mustangproject#566]
Hey,

with the newest Version (2.15.0) of Mustang we get an error. 
The example-PDf is from `www.ferd-net.de` in Version 2.3.2. It feels strange, that the pdf is valid despite the exceptions.

```
java -Xmx1G -Dfile.encoding=UTF-8 -jar Mustang-CLI-2.15.0.jar --no-notices --action validate --source ZF232_DE/Beispiele/4.\ EXTENDED/EXTENDED_Warenrechnung/EXTENDED_Warenrechnung.pdf 
10:39:16.361 [main] ERROR o.m.ZUGFeRD.ZUGFeRDInvoiceImporter - Failed to parse PDF
java.lang.RuntimeException: org.mustangproject.Exceptions.ArithmetricException: Could not reproduce the invoice, this could mean that it could not be read properly
	at org.mustangproject.ZUGFeRD.ZUGFeRDInvoiceImporter.setDocument(ZUGFeRDInvoiceImporter.java:271)
	at org.mustangproject.ZUGFeRD.ZUGFeRDInvoiceImporter.setRawXML(ZUGFeRDInvoiceImporter.java:240)
	at org.mustangproject.ZUGFeRD.ZUGFeRDInvoiceImporter.setRawXML(ZUGFeRDInvoiceImporter.java:254)
	at org.mustangproject.ZUGFeRD.ZUGFeRDInvoiceImporter.extractFiles(ZUGFeRDInvoiceImporter.java:215)
	at org.mustangproject.ZUGFeRD.ZUGFeRDInvoiceImporter.extractLowLevel(ZUGFeRDInvoiceImporter.java:150)
	at org.mustangproject.ZUGFeRD.ZUGFeRDInvoiceImporter.setInputStream(ZUGFeRDInvoiceImporter.java:95)
	at org.mustangproject.ZUGFeRD.ZUGFeRDInvoiceImporter.<init>(ZUGFeRDInvoiceImporter.java:81)
	at org.mustangproject.ZUGFeRD.ZUGFeRDImporter.<init>(ZUGFeRDImporter.java:43)
	at org.mustangproject.validator.PDFValidator.validate(PDFValidator.java:125)
	at org.mustangproject.validator.ZUGFeRDValidator.internalValidate(ZUGFeRDValidator.java:126)
	at org.mustangproject.validator.ZUGFeRDValidator.validate(ZUGFeRDValidator.java:240)
	at org.mustangproject.commandline.Main.performValidate(Main.java:469)
	at org.mustangproject.commandline.Main.main(Main.java:430)
Caused by: org.mustangproject.Exceptions.ArithmetricException: Could not reproduce the invoice, this could mean that it could not be read properly
	at org.mustangproject.ZUGFeRD.ZUGFeRDInvoiceImporter.extractInto(ZUGFeRDInvoiceImporter.java:745)
	at org.mustangproject.ZUGFeRD.ZUGFeRDInvoiceImporter.setDocument(ZUGFeRDInvoiceImporter.java:267)
	... 12 common frames omitted
10:39:16.458 [main] INFO  c.h.s.x.SchematronResourceXSLTCache - Compiling XSLT instance [cpPath=/xslt/ZF_232/FACTUR-X_EXTENDED.xslt; urlResolved=true; URL=jar:file:Mustang-CLI-2.15.0.jar!/xslt/ZF_232/FACTUR-X_EXTENDED.xslt]
10:39:18.539 [main] INFO  o.m.validator.ZUGFeRDValidator - Parsed PDF:valid XML:valid Signature:null Checksum:EE705D0C4E811287EE8C8B4E51E016E4BE0C5AB3 Profile:urn:cen.eu:en16931:2017#conformant#urn:factur-x.eu:1p0:extended Version:2 Took:3187ms Errors:[] 
<?xml version="1.0" encoding="UTF-8"?>

<validation filename="EXTENDED_Warenrechnung.pdf" datetime="2024-11-20 10:39:15">
  <pdf>ValidationResult [flavour=3b, totalAssertions=31178, assertions=[], isCompliant=true]
    <info>
      <signature>unknown</signature>
      <duration unit="ms">1019</duration>
    </info>
    <summary status="valid"/>
  </pdf>  
  <xml>
    <info>
      <version>2</version>
      <profile>urn:cen.eu:en16931:2017#conformant#urn:factur-x.eu:1p0:extended</profile>
      <validator version="2.15.0"/>
      <rules>
        <fired>255</fired>
        <failed>0</failed>
      </rules>
      <duration unit="ms">2138</duration>
    </info>
    <summary status="valid"/>
  </xml>
  <summary status="valid"/>
</validation>
``` 

The ZugferdImporter had a different handling now. If we ignore the Exception (see above), we got a Nullpointer now on some methods. In my case its `getDueDate()` and `getIBAN()` . With 2.14.2 the string was empty - like `getPaidAmount()`. Can you please fix it or do you need more information ?

```java
    
      ZUGFeRDImporter importer = new ZUGFeRDImporter(is); // Zugferd PDF-File

      importer.getDueDate(); // nullpointer since 2.15.0 ; previous empty 
      importer.getIBAN(); // nullpointer since 2.15.0; previous empty

      importer.getPaidAmount(); // empty string

``` 
Best Regards

**Comment by jstaerk** [comment]
Thanks for the report, the exception is now fixed  [2f4011c]. 

The cause for both issues is #435, which wasnt very well communicated: We used to use a separate parsing for the old convenient ZUGFeRDImporter and the much more complete new ZUGFeRDInvoiceImporter. This no longer works with UBL as input because we would have to maintain CII->Importert, CII->InvoiceImporter als well as UBL->Importer and UBL->InvoiceImporter mappings. Now we base the Importer on the invoiceImportert and only need to maintain CII->InvoiceImporter and UBL->InvoiceImporter.

As to the attributes: If they are not present(?), my point would be that it was previously wrong to return a "", it should always has been a null. If they are there, or if you think thats wrong, please open a separate ticket.

**Comment by BrooklynFFM** [comment]
I encountered the error after switching from 2.15.0 to 2.15.1. Could this be a regession?
